server-netty/src/main/java/org/jboss/aerogear/webpush/netty/WebPushFrameListener.java

```diff
@@ -98,10 +98,11 @@ public void onHeadersRead(final ChannelHandlerContext ctx,
                               final int padding,
                               final boolean endStream) throws Http2Exception {
         final String path = headers.path().toString();
-        LOGGER.info("onHeadersRead. streamId={}, method={}, path={}, endstream={}", streamId, headers.method(), path, endStream);
+        final String method = headers.method().toString();
+        LOGGER.info("onHeadersRead. streamId={}, method={}, path={}, endstream={}", streamId, method, path, endStream);
 
-        setPathAndMethod(encoder.connection().stream(streamId), path, headers.method());
-        switch (headers.method().toString()) {
+        streamPaths.put(streamId, path);
```

Comment:
In your constructor, create a Key via connection().newKey(). Then use that as the key in `getProperty` and `setProperty` instead of "webpush.path".